### ISSUE TITLE
 No OpKernel was registered to support Op 'RFFT'  for CPU (running on android)

### ISSUE THREAD

**Issue by liuzqt** [tensorflow/tensorflow#11804]
I'm running tensorflow on android. and got this error:
`Caused by: java.lang.IllegalArgumentException: No OpKernel was registered to support Op 'RFFT' with these attrs.  Registered devices: [CPU], Registered kernels:
                                                                       <no registered kernels>`

I'm using the master branch, where the RFFT CPU verison is already supported. So I'm wondering why this problem show up.

The TF version I used to build the graph(*.pb file) is also the latest master branch.
Here is to code I wrtie .pb graph:

        with tf.Graph().as_default(), tf.Session(config=tf.ConfigProto(
                allow_soft_placement=True)) as session:
            with tf.variable_scope("model"):
                model = DeployModel(config=config)

            print('Graph build finished')
            # variable_names = [n.name for n in
            #                   tf.get_default_graph().as_graph_def().node]
            # for n in variable_names:
            #     print(n)

            saver = tf.train.Saver()
            saver.restore(session, save_path=path_join(self.config.model_path,
                                                       'latest.ckpt'))
            print("model restored from %s" % config.model_path)

            frozen_graph_def = graph_util.convert_variables_to_constants(
                session, session.graph.as_graph_def(),
                ['model/inputX', 'model/softmax', 'model/nn_outputs'])
            tf.train.write_graph(
                frozen_graph_def,
                os.path.dirname(graph_path),
                os.path.basename(graph_path),
                as_text=False,
            )

So maybe RFFT for CPU is still not supported on android in the latest branch?

**Comment by aselle** [comment]
Android only builds a subset of the ops, You need to add the ops that you need that are not in the "commonly used set" by hacking the build files for your needs.

**Comment by andrewharp** [comment]
@liuzqt You can follow @aselle's suggestion by adding fft_ops.cc to the //tensorflow/core/kernels:android_extended_ops_group1 filegroup and see if that solves the issue.

**Comment by liuzqt** [comment]
@andrewharp @aselle adding fft_ops.cc to build file solve the problem, many thanks!  But it come up with a new problem, when running fft, raise this exception:
`Caused by: java.lang.IllegalArgumentException: DataType 8 is not recognized in Java (version 1.3.0-rc0)
                                                       at org.tensorflow.DataType.fromC(DataType.java:66)
                                                       at org.tensorflow.Tensor.fromHandle(Tensor.java:456)
                                                       at org.tensorflow.Session$Runner.runHelper(Session.java:312)
                                                       at org.tensorflow.Session$Runner.run(Session.java:245)`
when I refered to https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/main/java/org/tensorflow/DataType.java

found that DataType 8 is missing in this definition.

I guess I have to define this type so that java can recognize it. But what is DataType8, and the corresponding data type in java?

**Comment by aselle** [comment]
It seems like complexabs is not implemented. You could just try removing the #if !defined and see if it works. I'm not sure why that check was put in, but it is possible that it was for an old version that didn't support the complex type. @petewarden , any ideas?

**Comment by liuzqt** [comment]
@aselle I remove that #if !defined and it works! many thank!
